Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/ImportOneProcessController.java

```diff
@@ -154,6 +173,76 @@ public void importProcess(final String domain, final String owner)
     }
   }
 
+  private void importLinkedSubProcesses(final ProcessSummaryType importedModel, final BPMNDiagram bpmnDiagram,
+                                  final String domain, final String owner, final String name, final int folderId)
+      throws Exception {
+    Map<SubProcess, ProcessSummaryType> subProcessMap = new HashMap<>();
+    Map<ProcessSummaryType, Map<BPMNNode, BPMNNode>> subProcessModelClonedNodesMap = new HashMap<>();
+    int count = 0;
+    for (SubProcess subProcess : bpmnDiagram.getSubProcesses()) {
+      BPMNDiagram subProcessDiagram = BPMNDiagramFactory.newBPMNDiagram("");
+      Map<BPMNNode, BPMNNode> nodeMap = subProcessDiagram.cloneSubProcessContents(subProcess);
+      String subProcessName = name + "_subprocess" + ++count;
+      ProcessSummaryType subProcessSummary = importSubProcess(subProcessDiagram, domain, owner, subProcessName, folderId);
+      subProcessMap.put(subProcess, subProcessSummary);
+      subProcessModelClonedNodesMap.put(subProcessSummary, nodeMap);
+    }
```

Comment:
As discussed, we need to make this code easier to understand by using better naming, or it could be better with explicit class/data type rather than all using maps.

Reply:
A tree structure has been introduced to make this part easier to understand.